In [3]:
import numpy as np
import matplotlib as plt

In [33]:
%%time
n = 400000
tsum = 0
h = 2 / n
for k in range(0, n):
    x_k = -1 + h * k
    y_k = np.sqrt(1 - (x_k)**2)
    tsum += y_k * h
    #i = np.sum(h * y_k)
print(tsum)

per_difference = (tsum - (np.pi / 2)) / (np.pi / 2) *100
print("this yields a percent difference of:" , per_difference)

1.5707963202209734
this yields a percent difference of: -4.1850894807021146e-07
Wall time: 944 ms


a) When I ran it with only 100 slices, it produced a result of "1.5691342555492505." this yields a percent different of about 0.11%

b) while running the time magic, the closest I could get to 1 second was about 400,000 slices, which produced the result "1.5707963202209734", which is accurate up to the eighth decimal place.

In [7]:
#disregard this text, I just want to keep it to look back on

#%%time
#n = 100
#h = 2 / (n)
#x_k = -1 + h * k
#y_k = np.sqrt(1 - (x_k**2))
#k = 1
#while k < n:
    
#    i = np.sum(h * y_k) * n
#    k += 1 
#print(i)

0.3979949748426484
Wall time: 0 ns
